### PR TITLE
[Feature] Full angle range

### LINKED ISSUE #386: How to expand the angle range to [0, 360) ?
as the title

### LINKED ISSUE #902: Acquisition at 360° of detection angle
### What's the feature?

Currently there are Doc/Dle135/Dle90 angle detection methods, but is there a function that can detect angles in 360°?
It would be useful to be able to detect the top and bottom of an object at the same time.

### Any other context?

_No response_

### PR DESCRIPTION
**Pull request by calcoloergosum**
**NOTE**: This PR alone does not make training work.

Thank you for your amazing codebase.

## Motivation

Resolve #386  #565 #902 #1045 . Sometimes we want the full range of angle.

## Modification

- Added new `angle_version` called "full360".

## Use cases (Optional)

Anything that is not rotation invariant shaped objects.
For example:
- letters
- animal with head/tail. [Yudo](https://arxiv.org/pdf/2308.04542.pdf)
- arrows
- ...

## Note

- Introducing full angle range requires specific convention regarding the order of points in a bounding box.
For code in this PR, bounding box is defined to be the clockwise - `(lt, rt, br, bl)` or `[(-,-), (-,+), (+,+), (+, -)]`.
- First time contributing. Please let me know if I am doing something off.
- To make it work, `RBBoxAssigner` should be modified accordingly.

## Checklist

- [x] Pre-commit or other linting tools are used to fix the potential lint issues.
- [x] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by calcoloergosum** [comment]
I just realized that this repo already is working on [360 rotation detection](https://github.com/open-mmlab/mmrotate/tree/dev-1.x/projects/RR360) :P

**Comment by yangxue0827** [comment]
I think your implementation is better and more flexible.

**Comment by zytx121** [comment]
@calcoloergosum Thank you for your contribution. We will incorporate your angle representation into the project as one of the default angle representations.

File: mmrotate/datasets/pipelines/transforms.py [hunk 1]
```diff
@@ -80,8 +80,10 @@ def bbox_flip(self, bboxes, img_shape, direction):
         flipped = bboxes.copy()
         if direction == 'horizontal':
             flipped[:, 0] = img_shape[1] - bboxes[:, 0] - 1
+            flipped[:4] = flipped[[1, 0, 3, 2]].copy()
```
Comment (yangxue0827) [inline mmrotate/datasets/pipelines/transforms.py:83]: This new line of code will produce incorrect image flips, especially for the original centered representation. Please check it and fix it.

File: mmrotate/datasets/pipelines/transforms.py [hunk 2]
```diff
@@ -80,8 +80,10 @@ def bbox_flip(self, bboxes, img_shape, direction):
...
         elif direction == 'vertical':
             flipped[:, 1] = img_shape[0] - bboxes[:, 1] - 1
+            flipped[:4] = flipped[[1, 0, 3, 2]].copy()
```
Comment (yangxue0827) [inline mmrotate/datasets/pipelines/transforms.py:86]: This new line of code will produce incorrect image flips, especially for the original centered representation. Please check it and fix it.

**Pull request merged**